tests/std/tests/P1135R6_atomic_wait/test.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _ENABLE_ATOMIC_ALIGNMENT_FIX
+
+#include <atomic>
+#include <cassert>
+#include <chrono>
+#include <thread>
+
+template <class UnderlyingType>
+void test_atomic_wait_func(
+    UnderlyingType old_value, UnderlyingType new_value, std::chrono::steady_clock::duration waiting_duration) {
+
+    constexpr std::ptrdiff_t seq_max_size = 10;
+    char seq[seq_max_size + 1];
+    std::atomic<char*> base = seq;
+    auto add_seq            = [&](char ch) {
+        char* p = base.fetch_add(1, std::memory_order_relaxed);
+        assert(p - seq < seq_max_size);
+        *p = ch;
+    };
+
+    std::atomic<UnderlyingType> a{old_value};
+    a.wait(new_value);
+
+    add_seq('1');
+
+    std::thread thd([&] {
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('2');
+        a.notify_all();
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('3');
+        a.store(old_value);
+        a.notify_one();
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('4');
+        a.store(new_value);
+        a.notify_one();
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('6');
+    });
+
+    a.wait(old_value);
+    assert(a.load() == new_value);
+
+    add_seq('5');
+
+    thd.join();
+
+    add_seq('\0');
+    assert(strcmp(seq, "123456") == 0);
+}
+
+int main() {
+    auto duration = std::chrono::milliseconds(200);

```

Comment:
We conventionally avoid using `auto` in this manner. This can also be constexpr:

```suggestion
    constexpr std::chrono::milliseconds duration(200);
```

I'd also recommend choosing another name; although you aren't `using namespace std;`, when I see `duration` I immediately think of `std::duration`. `waiting_duration` would be fine.